yatima-inc/Wasm.lean
# Problem: No way to auto-test binary compatibility

Solution:
 - Never build full YSL, only stuff we need
 - Complete Dependent.lean test which tests for all the dependent functionality we have: names, numbers, digits.
    - They check that the hex digits are parsed correctly
    - They check that the chars that lie beyond the hex digits are not parsed
 - Move semi-auto Leb128 tests to their own file.
 - Test:
    - That special cases of leb128 conversions are correct.
    - Find a bug in Main.lean, fix it and check it in the test. - Url to a leb playground is: https://npm.runkit.com/%40webassemblyjs%2Fleb128 (you're welcome).
 - Add a test suite for simple Encodings
 - Add a new test file that will run the binary compatibility tests.
 - Make our own cmd running function that is capable of running stuff and grabbing the result into homebrew data structures.
 - Show how to run wasm-sandbox and verify that we can do it.
 - Verify by branching over the invocations of lspecIO.
 - Write a message for when tests fail that the user of this library has to provision wasm-sandbox binary into the current directory in order to run the tests. - It really is up to them how to get the binary.
 - Gitignore all the artefacts.
 - Finally, test all the modules we used to test in Main, but also test for binary compatibility!
 - Some tests fail, but it's ok. After all most of the binary encodings weren't tested, really.
 - We're including this patch into main and will go on to fix the errors shortly.

sq
